src/Business Foundation/App/NoSeries/src/Batch/NoSeriesBatchImpl.Codeunit.al

```diff
@@ -0,0 +1,226 @@
+    var
+        NoSeries: Codeunit "No. Series";
+    begin
+        SetInitialState(TempNoSeriesLine);
+        exit(NoSeries.PeekNextNo(TempGlobalNoSeriesLine, UsageDate));
+    end;
+
+    procedure GetNextNo(NoSeriesCode: Code[20]; UsageDate: Date; HideErrorsAndWarnings: Boolean): Code[20]
+    var
+        TempNoSeriesLine: Record "No. Series Line" temporary;
+    begin
+        GetNoSeriesLine(TempNoSeriesLine, NoSeriesCode, UsageDate, HideErrorsAndWarnings);
+        exit(GetNextNo(TempNoSeriesLine, UsageDate, HideErrorsAndWarnings));
+    end;
+
+    procedure GetNextNo(var TempNoSeriesLine: Record "No. Series Line" temporary; UsageDate: Date; HideErrorsAndWarnings: Boolean): Code[20]
+    var
+        NoSeries: Codeunit "No. Series";
+        NextNo: Code[20];
+    begin
+        SyncGlobalLineWithProvidedLine(TempNoSeriesLine, UsageDate);
+        LockedNoSeriesLine.LockTable();
+        NextNo := NoSeries.GetNextNo(TempGlobalNoSeriesLine, UsageDate, HideErrorsAndWarnings);
+        TempNoSeriesLine := TempGlobalNoSeriesLine;
+        exit(NextNo);
+    end;
+
+    procedure SimulateGetNextNo(NoSeriesCode: Code[20]; UsageDate: Date; PrevDocumentNo: Code[20]): Code[20]
+    var
+        TempNoSeriesLine: Record "No. Series Line" temporary;
+        NoSeries: Codeunit "No. Series";
+        NoSeriesStatelessImpl: Codeunit "No. Series - Stateless Impl.";
+    begin
+        if NoSeriesCode = '' then
+            exit(IncStr(PrevDocumentNo));
+
+        SetSimulationMode();
+
+        GetNoSeriesLine(TempNoSeriesLine, NoSeriesCode, UsageDate);
+        if NoSeries.MayProduceGaps(TempNoSeriesLine) then
+            TempNoSeriesLine.Implementation := TempNoSeriesLine.Implementation::Normal;
+        TempNoSeriesLine."Last No. Used" := PrevDocumentNo;
+
+        if not NoSeriesStatelessImpl.EnsureLastNoUsedIsWithinValidRange(TempNoSeriesLine, true) then
+            exit(IncStr(PrevDocumentNo));
+
+        TempNoSeriesLine.Modify(false);
+        exit(GetNextNo(TempNoSeriesLine, UsageDate, false));
+    end;
+
+    procedure GetLastNoUsed(NoSeriesCode: Code[20]): Code[20]
+    var
+        TempNoSeriesLine: Record "No. Series Line" temporary;
+        NoSeries: Codeunit "No. Series";
+    begin
+        if not GetNoSeriesLine(TempNoSeriesLine, NoSeriesCode, WorkDate(), true) then
+            exit('');
+        exit(NoSeries.GetLastNoUsed(TempGlobalNoSeriesLine));
+    end;
+
+    procedure GetLastNoUsed(TempNoSeriesLine: Record "No. Series Line" temporary): Code[20]
+    var
+        NoSeries: Codeunit "No. Series";
+    begin
+        SyncGlobalLineWithProvidedLine(TempNoSeriesLine, TempNoSeriesLine."Starting Date");
+        exit(NoSeries.GetLastNoUsed(TempGlobalNoSeriesLine));
+    end;
+
+    procedure SetSimulationMode()
+    begin
+        SimulationMode := true;
+    end;
+
+    [InherentPermissions(PermissionObjectType::TableData, Database::"No. Series Line", 'rm', InherentPermissionsScope::Both)]
```

Comment:
Default for inherentpermissionscope is both. https://learn.microsoft.com/en-us/dynamics365/business-central/dev-itpro/developer/methods-auto/inherentpermissionsscope/inherentpermissionsscope-option